frontend/src/tests/user-permission-tests.js

```diff
@@ -14,42 +14,127 @@ import { databaseSeeder } from "./setup";
 export function userPermissionsTests(api) {
     const { apiGET, apiPOST } = api;
 
-    const newUserData = {
+    let defaultUser;
+
+    const newInstructorOnlyUserData = {
         utorid: "userxx",
         roles: ["instructor"],
     };
 
+    const newTAOnlyUserData = {
+        utorid: "tauser",
+        roles: ["ta"],
+    };
+
+    const newTAInstructorUserData = {
+        utorid: "tainstuser",
+        roles: ["ta", "instructor"],
+    };
+
+    /**
+     * Switch current active user to the user represented by the given user data. If given user does not exist, one will
+     * be created.
+     * @param userData the user to be set to the active user
+     * @returns {Promise<void>}
+     */
+    async function switchToTargetUser(userData) {
+        let resp = await apiPOST("/debug/users", userData);
```

Comment:
You create a new user each time before you switch to it? That doesn't seem right...